### DISCUSSION TITLE
Company mode and JavaScript (can't find `company-js`)

### DISCUSSION THREAD

**Discussion by fergiemcdowall** [company-mode/company-mode#1424] [Q&A] [answered]
Hi! I was just wondering why there doesn't seem to be a `company-js` or `company-javascript` package? Or to put it another way- if you want to get `company` working on .js files, what is the best/recommended approach?

(apologies if this is answered already- a search for "javascript" in discussion did not yield helpful results)

**Comment by fergiemcdowall** [comment]
Update: the vanilla `company` package seems to work pretty well for JavaScript. That being the case- what extra functionality do the language specific company packages  (`company-<lang>`) generally provide for their respective languages?.

**Comment by dgutov** [comment, marked as answer]
Hi! Only a minority of the built-in backends are language-specific ones. And we often try to migrate those into the respective major modes (as has already happened with company-elisp and company-css).

For JavaScript, you can keep using one of the built-in ones (dabbrev-code or etags) if it's a smaller project and/or you know how to generate a TAGS file (and like that approach, which it not everyone's cup of tea), or install one of the available LSP clients for richer experience (eglot or lsp-mode): they plug into Company for completion UI.

**Reply by dgutov** [reply]
Note that using an LSP client also comes with the requirement to install an appropriate language server.

**Reply by fergiemcdowall** [reply]
Ok- thanks for the explanation and the pointers
